Sources/Core/Views/ProfileImageView/ProfileImageView.swift

```diff
@@ -0,0 +1,168 @@
+//
+//  ProfileImageView.swift
+//  PovioKit
+//
+//  Created by Arlind Dushi on 3/9/22.
+//  Copyright © 2022 Povio Inc. All rights reserved.
+//
+
+import SwiftUI
+
+@available(iOS 13, *)
+public class ProfileImageProperties: ObservableObject {
+  @Published public var placeHolder = Image(systemName: "person")
+  @Published public var backgroundType = ProfileImageView.BackgroundType.plain(.clear)
+  @Published public var cornerRadius: CGFloat = 0
+  @Published public var contentMode: ContentMode = .fit
+  @Published public var borderColor: Color = .clear
+  @Published public var borderWidth: CGFloat = 0
+  @Published public var badging: ProfileImageView.BadgingMode = .none
+  @Published public var badgeAlignment: Alignment = .bottomTrailing
+  @Published var urlData: Data?
+}
+
+@available(iOS 13, *)
+public struct ProfileImageView: View {
+  public var imageTapped: (() -> Void)?
+  public var badgeTapped: (() -> Void)?
+  @ObservedObject public var properties = ProfileImageProperties()
+  
+  public struct BadgeStyle {
+    let image: Image
+    let backgroundColor: Color
+    let borderColor: Color?
+    let borderWidth: CGFloat?
+  }
+  
+  public enum BadgingMode {
+    case none
+    case some(badge: BadgeStyle)
+  }
+  
+  public enum BackgroundType {
+    case plain(Color)
+    case linearGradient(LinearGradient)
+    case radialGradient(RadialGradient)
+    case angularGradient(AngularGradient)
+  }
+  
+  public init() {}
+  
+  public var body: some View {
+    GeometryReader { geo in
+      ZStack(alignment: properties.badgeAlignment) {
+        ProfileImage(properties: properties, profileTapped: triggerImageTapClosure)
+        Badge(style: properties.badging, size: .init(width: geo.size.width / 3.5, height: geo.size.height / 3.5), badgeTapped: tiggerBadgeTapClosure)
+      }
+    }
+  }
+  
+  private func triggerImageTapClosure() { imageTapped?() }
+  private func tiggerBadgeTapClosure() { badgeTapped?() }
+  
+  private struct Badge: View {
```

Comment:
Under extension? Also, rename to `BadgeView` 🤷‍♂️

Reply:
Maybe we can use `Delegated`, which automatically handles reference cycles, instead of plain closures for callbacks.